### ISSUE TITLE
Outcome evaluation problem

### ISSUE THREAD

**Issue by tang12138-yyhh** [eml-eda/mixprec-pruning#4]
Hello, after I have adjusted the regularization strength parameter many times, there are still some problems:
1. When I set the regularization intensity to 1e-5, the result is as follows:
![image](https://github.com/user-attachments/assets/8d171374-5f39-4a08-9c2c-5fdb663e3fa7)
2. The result has not been ideal, and the accuracy is basically around 60%. Could you please provide me with your hyperparameter setting for reference?
3. What does the model size here: Tensor (28946) refer to? The number of parameters? The units don't look quite like the picture in your paper.
The chart below shows the results of the paper：
![image](https://github.com/user-attachments/assets/2dc93041-d78b-46ba-8300-cc7c4f2379ec)
In my opinion, on the leftmost graph, each point in the graph represents a result of regularization intensity (Test Accuracy + Size), where the size unit is (size), what is the size unit in the log file and are these two equivalent?

**Comment by bmots** [comment]
Hello @tang12138-yyhh ,

(1. + 2.) That result is due to a too high value set for the regularization strength. You can try with smaller values, for instance in the range [1e-8, 1e-6]. The lower the value of this hyperparameter, the more importance is assigned to the task loss with respect to the regularization loss in the training phase, thus encouraging the convergence to more accurate models, at the cost of a higher complexity (e.g. higher number of parameters).

(3.) As you can see in the plot, the unit of measurement is the kilobyte (kB). The logged model size counts the number of bits required to store the weights of the model, so the unit of measurement is the bit. Thus, a rescaling is necessary to map the values in the log to kB. After this step, you can see that your result is compatible with the ones shown in the plot.

**Comment by tang12138-yyhh** [comment]
Thanks, @bmots ,
So what does his scaling factor look like? How do I scale the parameter quantity to KB?

**Comment by bmots** [comment]
Hello @tang12138-yyhh ,
As stated in the previous reply, it's a conversion from bits to kilobytes.

**Comment by tang12138-yyhh** [comment]
Hello, @bmots ,
<img width="421" alt="image" src="https://github.com/user-attachments/assets/48cbf85b-8208-4aaf-bf15-24f730328800">

I wonder where the seed model size comes from here?

**Comment by bmots** [comment]
Hello @tang12138-yyhh ,
it is the size in kilobytes of the seed architecture, i.e. the architecture that we want to optimize (imported and created [here](https://github.com/eml-eda/mixprec-pruning/blob/main/icl_mixprec_training_cost_size.py#L213) in the code). Its size has no dependency on our optimization procedure, you can compute it as for any "standard" neural network.
